File: java/src/main/java/com/bookstore/DiscountVoucher.java
package com.bookstore;

public class DiscountVoucher implements CartAble {

    private final int percentage;
    private final PutIntoCart putIntoCart = new PutItemIntoCart();

    public DiscountVoucher(int percentage) {
        this.percentage = percentage;
    }

    @Override
    public void putIntoMyCart(Cart cart) {
        putIntoCart.putIntoCart(cart, () -> "Discount voucher for " + percentage + "% on your next order");
    }

}

File: csharp/BookStore/src/BookStore/DiscountVoucher.cs
namespace BookStore
{
    public class DiscountVoucher : CartAble
    {
        private readonly int _percentage;

        public DiscountVoucher(int percentage)
        {
            _percentage = percentage;
        }

        public void PutIntoMyCart(Cart cart)
        {
            cart.Add($"Discount voucher for {_percentage}% on your next order");
        }
    }
}
